File: katas/UserSignUp/src/core/backend/domain/Name.test.ts
import {Name} from "./Name";
/** Approach: Inside-out
 * 1.- Name is created
 * 2.- Name cannot be empty. 
*/

describe('Name test.', () => {

  test('Name is created', () => {
    const myName: Name = Name.createName("Nico");
    expect(myName).toBeInstanceOf(Name)
  });

  test('Name cannot be empty', () => {
    expect(() => Name.createName("") ).toThrow("Name cannot be empty.")
  });


})

File: ejercicios/5_DomainDriverDesign-Chapter-3/src/main/java/org/layers/domain/value_objects/NameTest.java
package org.layers.domain.value_objects;

import io.vavr.control.Either;
import org.layers.domain.warnings.NameWarning;
import org.junit.jupiter.api.*;

import static org.junit.jupiter.api.Assertions.*;

public class NameTest {
    /** Use cases:
     * 1. Name should be created.
     * 2. Empty name -> return NameWarning.NAME_CANNOT_BE_EMPTY
     */
    String normal_name;
    String emptyName;
    @BeforeEach
    void setUp() {
        normal_name = "Nico";
        emptyName = "";
    }
    @Test
    public void nameShouldBeCreated() {
        Either<NameWarning,Name> nameCreated = Name.createName(normal_name);
        Name name = nameCreated.get();

        assertEquals(name.getName(), normal_name);
    }
    @Test
    public void emptyNameShouldNotBeCreated() {
        Either<NameWarning ,Name> nameCreated = Name.createName(emptyName);

        assertEquals(nameCreated.getLeft(), NameWarning.NAME_CANNOT_BE_EMPTY);
    }
}
